### PR TITLE
[SYCL][NFC] Fix ac_float test for compilation with FE optimizations

### PR DESCRIPTION
**Pull request by mlychkov**
Signed-off-by: Mikhail Lychkov <mikhail.lychkov@intel.com>

### CODE REVIEW

File: sycl/test/fpga_tests/fpga_ihs_float.cpp [hunk 1]
```diff
@@ -18,7 +18,7 @@ void ap_float_cast() {
   ap_int<1 + Eout + Mout> float_cast_res =
       __spirv_ArbitraryFloatCastINTEL<1 + EA + MA, 1 + Eout + Mout>(
           A, MA, Mout, Subnorm, RndMode, RndAcc);
-  // CHECK: %{{.*}} = call spir_func i40 @_Z{{[0-9]+}}__spirv_ArbitraryFloatCastINTEL{{.*}}(i40 %{{[0-9]+}}, i32 28, i32 30, i32 0, i32 2, i32 1)
+  // CHECK: call spir_func i40 @_Z{{[0-9]+}}__spirv_ArbitraryFloatCastINTEL{{.*}}(i40 {{[%a-z0-9.]+}}, i32 28, i32 30, i32 0, i32 2, i32 1)
```
Comment (bader) [inline sycl/test/fpga_tests/fpga_ihs_float.cpp:21]: Please, add following command:
```
// RUN: %clangxx -I %sycl_include -S -emit-llvm -fsycl -fsycl-device-only %s -Xclang -fsycl-enable-optimizations -o - | FileCheck %s
```
Last time I checked, enabling optimizations emits empty LLVM module as this code has no side effects.

Comment (mlychkov) [reply sycl/test/fpga_tests/fpga_ihs_float.cpp:21]: Done.

Comment (bader) [reply sycl/test/fpga_tests/fpga_ihs_float.cpp:21]: I'm not sure, but you might want to leave original run command as well.

**Comment by MrSidims** [comment]
I don't get it. I was thinking that we are getting an empty module with the optimizations enable and hence we need to add side effects to the functions in the test.  From what I see, this patch just changes CHECK strings, so the only thing that has changed is variable names?

Please also move the test to `check_device_code`, I think it fits to it better than to 'fpga_tests'.

**Comment by mlychkov** [comment]
> I don't get it. I was thinking that we are getting an empty module with the optimizations enable and hence we need to add side effects to the functions in the test. From what I see, this patch just changes CHECK strings, so the only thing that has changed is variable names?

Yes, calls are still present, only differ a bit from unoptimized version.

> Please also move the test to `check_device_code`, I think it fits to it better than to 'fpga_tests'.

Moved this and ac_fixed tests into  `check_device_code` dir.

**Pull request merged**